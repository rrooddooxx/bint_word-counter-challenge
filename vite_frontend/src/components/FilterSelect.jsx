import Form from "react-bootstrap/Form";
import PropTypes, { array } from "prop-types";

const FilterSelect = ({ optLabel, options, onChangeFn }) => {
  return (
    <div className="input-group">
      <label className="input-group-text" htmlFor="inputGroupSelect01">
        {optLabel}
      </label>
      <Form.Select
        aria-label="select-dropdown"
        id="inputGroupSelect01"
        onChange={e => onChangeFn(parseInt(e.target.value))}
      >
        <option value={0}>Seleccione...</option>
        {options.map(({ name, val }) => {
          return (
            <option value={val} key={name + val}>
              {name}
            </option>
          );
        })}
      </Form.Select>
    </div>
  );
};

FilterSelect.propTypes = {
  options: PropTypes.arrayOf(PropTypes.object),
  name: PropTypes.string,
  val: PropTypes.number,
  onChangeFn: PropTypes.func,
};

FilterSelect.defaultProps = {
  optLabel: "Opciones: ",
  options: [{}],
  onChangeFn: () => {},
};

export { FilterSelect };
