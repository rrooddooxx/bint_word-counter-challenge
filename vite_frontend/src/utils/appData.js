export const data = {
  limitResultsOptions: [
    { name: "Ver 15 Resultados", val: 15 },
    { name: "Ver 30 Resultados", val: 30 },
    { name: "Ver 50 Resultados", val: 50 },
  ],
  wordLengthOptions: [
    { name: "Desde 1 caracter", val: 1 },
    { name: "Desde 2 caracteres", val: 1 },
    { name: "Desde 3 caracteres", val: 2 },
  ],
  navbarSections: [
    { name: "Inicio", href: "/" },
    { name: "API Docs", href: "#apidocs" },
  ],
  apiUrl: "http://localhost:3000/api/v1/ranking",
};
