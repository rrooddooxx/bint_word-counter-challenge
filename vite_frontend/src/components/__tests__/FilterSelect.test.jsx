import { data } from "../../utils/appData.js";
import { fireEvent, render, screen } from "@testing-library/react";
import { FilterSelect } from "../FilterSelect.jsx";

describe("[COMP] <FilterSelect>", () => {
  const options = data.limitResultsOptions;
  test("Match con el snapshot pasándole array de props", () => {
    const { container } = render(
      <FilterSelect options={options} onChangeFn={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Muestra el mismo label de opciones que se le pasa mediante props", () => {
    const optLabel = "Densidad de palabra: ";
    render(
      <FilterSelect
        optLabel={optLabel}
        options={options}
        onChangeFn={() => {}}
      />
    );
    expect(screen.getByText("Densidad de palabra:").innerHTML).toBe(optLabel);
  });

  test("Adquiere el valor del arreglo de opciones pasado por props", () => {
    const options = [{ name: "once", val: 11 }];
    render(<FilterSelect options={options} onChangeFn={() => {}} />);
    const selectNode = screen.getByLabelText("select-dropdown");
    fireEvent.change(selectNode, { target: { value: 11 } });
    expect(selectNode.value).toBe(options[0].val.toString());
  });

  test("Llama a la función onChangeFn en el evento onChange, pasándole el valor del option seleccionado", () => {
    const onChangeFn = jest.fn();
    const optionsArr = [
      { name: "dos", val: 2 },
      { name: "tres", val: 3 },
    ];

    render(<FilterSelect options={optionsArr} onChangeFn={onChangeFn} />);
    const dropdownElement = screen.getByLabelText("select-dropdown");
    fireEvent.change(dropdownElement, { target: { value: 3 } });
    fireEvent.change(dropdownElement, { target: { value: 2 } });

    expect(onChangeFn).toHaveBeenCalledTimes(2);
    expect(onChangeFn).toHaveBeenCalledWith(2);
  });
});
