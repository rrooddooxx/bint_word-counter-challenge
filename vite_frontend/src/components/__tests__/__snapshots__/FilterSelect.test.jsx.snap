// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMP] <FilterSelect> Match con el snapshot pas√°ndole array de props 1`] = `
<div>
  <div
    class="input-group"
  >
    <label
      class="input-group-text"
      for="inputGroupSelect01"
    >
      Opciones: 
    </label>
    <select
      aria-label="select-dropdown"
      class="form-select"
      id="inputGroupSelect01"
    >
      <option
        value="0"
      >
        Seleccione...
      </option>
      <option
        value="15"
      >
        Ver 15 Resultados
      </option>
      <option
        value="30"
      >
        Ver 30 Resultados
      </option>
      <option
        value="50"
      >
        Ver 50 Resultados
      </option>
    </select>
  </div>
</div>
`;
